docs/evm/cadence/batched-evm-transactions.md

```diff
@@ -0,0 +1,641 @@
+
+:::
+
+### Approve ERC721 Transfer
+
+Our example `MaybeMintERC721` contract accepts WFLOW in exchange for minting an NFT. However, the contract cannot move
+WFLOW without your permission. To allow the contract to move your WFLOW, you must approve the contract to transfer
+enough of your WFLOW to mint the NFT.
+
+### Mint ERC721 Token
+
+Finally, we'll mint an ERC721 token using the `MaybeMintERC721` contract. This contract has a 50% chance of failing,
+simulating a real-world scenario where purchasing an NFT might fail - say a listing was purchased before your
+transaction was processed.
+
+Importantly, if this transaction fails, we want to revert the entire sequence of transactions. After all, you wrapped
+FLOW to WFLOW and approved the ERC721 transfer specifically to mint this NFT. If the mint fails, you want to unwind
+everything. This is where batching EVM transactions using Cadence is extremely powerful.
+
+## Interacting with the Contracts
+
+Before taking the easy route, let's first interact with the contracts individually to better understand the process and
+status quo user experience. Realistically, this is your only option for completing the whole process on other EVM
+platforms.
+
+:::tip
+
+Recall in [Prerequisites](#prerequisites) that you need to have both MetaMask and Flow Wallet installed and funded with
+Testnet FLOW. Make sure you've done so before proceeding.
+
+:::
+
+### Using MetaMask
+
+#### 1. Wrap FLOW
+
+Our first action will be to wrap enough FLOW to cover the cost of minting the `MaybeMintERC721` token. To do this, we'll
+interact with the `WFLOW` contract on Testnet. There are a number of ways we could interact with this contract - Remix
+IDE, Foundry's CLI, Hardhat, etc. - but for the purposes of this guide, we'll use the [Flowscan EVM block
+explorer](https://www.evm-testnet.flowscan.io/).
+
+Navigate to the WFLOW Testnet contract on Flowscan:
+[WFLOW](https://evm-testnet.flowscan.io/token/0xd3bF53DAC106A0290B0483EcBC89d40FcC961f3e?tab=write_contract). Ensure
+you're on the "Write Contract" tab which allows you to interact with the contract's mutating functions. 
+
+Before you can interact with the contract, you need to connect your MetaMask wallet to the Flowscan EVM block explorer.
+Click the "Connect" button in the top right corner and follow the prompts to connect your MetaMask wallet.
+
+![Connect wallet to Flowscan](./flowscan-connect.png)
+
+Once connected, you should see your address in the top right corner and above the contract's functions.
+
+Now we can wrap FLOW. Click on the `deposit` method which will drop down an input field for the amount of FLOW you want
+to wrap. The mint amount for the `MaybeMintERC721` contract is 1 whole FLOW which in EVM terms is `1e18 wei` - `wei`
+being the smallest unit of an EVM's native currency (inherited from Ethereum's units - more on Ether units
+[here](https://www.cyfrin.io/glossary/ether-and-wei-code-example)).
```

Comment:
I'd probably link to the Solidity docs here instead of a random vendor (unless we have a relationship with them)

https://docs.soliditylang.org/en/v0.8.28/units-and-global-variables.html#ether-units